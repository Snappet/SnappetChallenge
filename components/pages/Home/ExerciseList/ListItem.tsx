import React from 'react';
import styled from '@emotion/styled';
import { Card, CardContent, Grid } from '@material-ui/core';
import { format } from 'date-fns/fp';

import { Exercise } from '@lib/models/Exercise';
import LabelValue from '@components/LabelValue';
import { fluidRange } from 'polished';

const formatDate = (date: string) => {
  const parsedDate = new Date(date);

  return format('dd/MM/yyyy HH:mm bb', parsedDate);
};

interface ListItemProps {
  exercise: Exercise;
}

const ListItem: React.FC<ListItemProps> = ({ exercise }) => {
  return (
    <Container>
      <CardContent>
        <Grid container>
          <LabelValue xs={12} label="Student" value={exercise.UserId} />

          <LabelValue
            xs={7}
            md={6}
            lg={7}
            label="Exercise"
            value={exercise.ExerciseId}
          />
          <LabelValue
            xs={5}
            md={6}
            lg={5}
            label="Correct"
            value={exercise.Correct == 1 ? 'Yes' : 'No'}
          />

          <LabelValue
            xs={7}
            md={6}
            lg={7}
            label="Difficulty"
            value={parseFloat(exercise.Difficulty).toFixed(2)}
          />
          <LabelValue
            xs={5}
            md={6}
            lg={5}
            label="Progress"
            value={exercise.Progress}
          />

          <LabelValue xs={12} md={6} lg={7} label="Domain" value={exercise.Domain} />
          <LabelValue xs={12} md={6} lg={5} label="Subject" value={exercise.Subject} />

          <LabelValue
            xs={12}
            label="Learning Objective"
            value={exercise.LearningObjective}
          />

          <LabelValue
            xs={12}
            md={6}
            lg={7}
            label="Submitted Answer"
            value={exercise.SubmittedAnswerId}
          />
          <LabelValue
            xs={12}
            md={6}
            lg={5}
            label="Submitted At"
            value={formatDate(exercise.SubmitDateTime)}
          />
        </Grid>
      </CardContent>
    </Container>
  );
};

export default ListItem;

const Container = styled(Card)`
  margin: 8px auto;

  display: flex;
  flex-direction: column;

  height: calc(100% - 16px);

  ${fluidRange({
    prop: 'width',
    fromSize: '312px',
    toSize: '656px',
  })};
`;
