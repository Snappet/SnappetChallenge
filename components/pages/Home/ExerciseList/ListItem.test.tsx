import React from 'react';
import { render } from '@testing-library/react';

import ListItem from './ListItem';

describe('ListItem', () => {
  it('matches snapshot', () => {
    const exercise = {
      SubmittedAnswerId: 2628622,
      SubmitDateTime: '2015-03-02T08:23:45.070',
      Correct: 1,
      Progress: 2,
      UserId: 40275,
      ExerciseId: 395191,
      Difficulty: '276.6952772',
      Subject: 'Subject Test 1',
      Domain: 'Domain Test 1',
      LearningObjective: 'Learning Objective 1',
    };

    const { container } = render(<ListItem exercise={exercise} />);
    expect(container).toMatchSnapshot();
  });
});
