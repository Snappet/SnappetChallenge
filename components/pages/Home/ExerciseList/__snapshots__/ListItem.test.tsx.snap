// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem matches snapshot 1`] = `
.emotion-0 {
  margin: 8px auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: calc(100% - 16px);
  width: 312px;
}

@media (min-width: 320px) {
  .emotion-0 {
    width: calc(186.91px + 39.09vw);
  }
}

@media (min-width: 1200px) {
  .emotion-0 {
    width: 656px;
  }
}

<div>
  <div
    class="MuiPaper-root MuiCard-root emotion-0 emotion-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiCardContent-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-item MuiGrid-align-items-xs-baseline MuiGrid-grid-xs-12"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
            >
              Student
              :
            </p>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              40275
            </p>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-item MuiGrid-align-items-xs-baseline MuiGrid-grid-xs-7 MuiGrid-grid-md-6 MuiGrid-grid-lg-7"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
            >
              Exercise
              :
            </p>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              395191
            </p>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-item MuiGrid-align-items-xs-baseline MuiGrid-grid-xs-5 MuiGrid-grid-md-6 MuiGrid-grid-lg-5"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
            >
              Correct
              :
            </p>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              Yes
            </p>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-item MuiGrid-align-items-xs-baseline MuiGrid-grid-xs-7 MuiGrid-grid-md-6 MuiGrid-grid-lg-7"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
            >
              Difficulty
              :
            </p>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              276.70
            </p>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-item MuiGrid-align-items-xs-baseline MuiGrid-grid-xs-5 MuiGrid-grid-md-6 MuiGrid-grid-lg-5"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
            >
              Progress
              :
            </p>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              2
            </p>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-item MuiGrid-align-items-xs-baseline MuiGrid-grid-xs-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-7"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
            >
              Domain
              :
            </p>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              Domain Test 1
            </p>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-item MuiGrid-align-items-xs-baseline MuiGrid-grid-xs-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-5"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
            >
              Subject
              :
            </p>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              Subject Test 1
            </p>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-item MuiGrid-align-items-xs-baseline MuiGrid-grid-xs-12"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
            >
              Learning Objective
              :
            </p>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              Learning Objective 1
            </p>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-item MuiGrid-align-items-xs-baseline MuiGrid-grid-xs-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-7"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
            >
              Submitted Answer
              :
            </p>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              2628622
            </p>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-item MuiGrid-align-items-xs-baseline MuiGrid-grid-xs-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-5"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
            >
              Submitted At
              :
            </p>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              02/03/2015 08:23 AM
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
